torch_geometric/nn/conv/hgt_conv.py

```diff
@@ -142,29 +148,94 @@ def forward(
         H, D = self.heads, self.out_channels // self.heads
 
         k_dict, q_dict, v_dict, out_dict = {}, {}, {}, {}
-
-        # Iterate over node-types:
-        for node_type, x in x_dict.items():
-            k_dict[node_type] = self.k_lin[node_type](x).view(-1, H, D)
-            q_dict[node_type] = self.q_lin[node_type](x).view(-1, H, D)
-            v_dict[node_type] = self.v_lin[node_type](x).view(-1, H, D)
-            out_dict[node_type] = []
-
-        # Iterate over edge-types:
-        for edge_type, edge_index in edge_index_dict.items():
-            src_type, _, dst_type = edge_type
-            edge_type = '__'.join(edge_type)
-
-            a_rel = self.a_rel[edge_type]
-            k = (k_dict[src_type].transpose(0, 1) @ a_rel).transpose(1, 0)
-
-            m_rel = self.m_rel[edge_type]
-            v = (v_dict[src_type].transpose(0, 1) @ m_rel).transpose(1, 0)
-
-            # propagate_type: (k: Tensor, q: Tensor, v: Tensor, rel: Tensor)
-            out = self.propagate(edge_index, k=k, q=q_dict[dst_type], v=v,
-                                 rel=self.p_rel[edge_type], size=None)
-            out_dict[dst_type].append(out)
+        if WITH_PYG_LIB:
```

Comment:
I am personally fine with making `PYG_LIB` mandatory here - I guess this avoids a lot of diverging code paths.